sound/soc/sof/Kconfig

```diff
@@ -87,6 +87,27 @@ config SND_SOC_SOF_STRICT_ABI_CHECKS
 	  If you are not involved in SOF releases and CI development
 	  select "N".
 
+choice
+	prompt "Virtualisation role"
+	default SND_SOC_SOF_VIRTIO_NONE
+	depends on VIRTIO
+	---help---
+	  If this kernel should be used in a virtualised environment, select
+	  which role it will be performing.
+
+config SND_SOC_SOF_VIRTIO_NONE
+	bool "None"
+	---help---
+	  No virtualisation
+
+config SND_SOC_SOF_VIRTIO_FE
+	bool "Guest"
+	depends on SND_SOC_SOF_NOCODEC_SUPPORT
+	---help---
```

Comment:
This should be a single enable option in kconfig to build a single binary that can be a VFE, VBE or regular driver depending on platform data.

Reply:
@lgirdwood you prefer a complete run-time detection? Building support for both and then checking if it's a virtual PCI ID - use the guest configuration, otherwise if VirtIO is enabled - also try to enable the virtual SOF host?

Reply:
Yes, runtime detection. VFE is based on PCI ID. VBE could be a seperate module that is loaded to enable the BE.

Reply:
reworked, you can now build SOF with support for both a host and a guest. Although, this wouldn't be very practical, since the guest depends on NOCODEC.